storage/netapp/netapp_e_storagepool.py

```diff
@@ -0,0 +1,888 @@
+                    data=json.dumps(secure_pool_data), headers=self.post_headers, method='POST',
+                    url_username=self.api_usr,
+                    url_password=self.api_pwd, validate_certs=self.validate_certs, timeout=120, ignore_errors=True)
+            except:
+                err = get_exception()
+                pool_id = self.pool_detail['id']
+                self.module.exit_json(
+                    msg="Failed to update storage pool. Pool id [%s]. Array id [%s].  Error[%s]." % (pool_id,
+                                                                                                     self.ssid,
+                                                                                                     str(err)))
+
+    @property
+    def needs_raid_level_migration(self):
+        current_raid_level = self.pool_detail['raidLevel']
+        needs_migration = self.raid_level != current_raid_level
+
+        if needs_migration:  # sanity check some things so we can fail early/check-mode
+            if current_raid_level == 'raidDiskPool':
+                self.module.fail_json(msg="raid level cannot be changed for disk pools")
+
+        return needs_migration
+
+    def migrate_raid_level(self):
+        self.debug("migrating storage pool to raid level '%s'..." % self.raid_level)
+        sp_raid_migrate_req = dict(
+            raidLevel=self.raid_level
+        )
+        try:
+            (rc, resp) = request(
+                self.api_url + "/storage-systems/%s/storage-pools/%s/raid-type-migration" % (self.ssid,
+                                                                                             self.name),
+                data=json.dumps(sp_raid_migrate_req), headers=self.post_headers, method='POST',
+                url_username=self.api_usr,
+                url_password=self.api_pwd, validate_certs=self.validate_certs, timeout=120)
+        except:
+            err = get_exception()
+            pool_id = self.pool_detail['id']
+            self.module.exit_json(
+                msg="Failed to change the raid level of storage pool. Pool id [%s]. Array id [%s].  Error[%s]." % (
+                    pool_id, self.ssid, str(err)))
+
+    @property
+    def sp_drives(self, exclude_hotspares=True):
+        if not self._sp_drives_cached:
+
+            self.debug("fetching drive list...")
+            try:
+                (rc, resp) = request(self.api_url + "/storage-systems/%s/drives" % (self.ssid), method='GET',
+                                     url_username=self.api_usr, url_password=self.api_pwd,
+                                     validate_certs=self.validate_certs)
+            except:
+                err = get_exception()
+                pool_id = self.pool_detail['id']
+                self.module.exit_json(
+                    msg="Failed to fetch disk drives. Array id [%s].  Error[%s]." % (pool_id, self.ssid, str(err)))
```

Comment:
Looks like too many args for the format string here  (3 args, but only 2 in the format string...)